### PR TITLE
Deprecation of NN, math, and tensor

### PR DESCRIPTION
**Pull request by owerbat**
Deprecation macros for NN, math, and tensor added

### CODE REVIEW

**Comment by owerbat** [comment]
make test

**Comment by owerbat** [comment]
make test

**Comment by owerbat** [comment]
make test

**Comment by owerbat** [comment]
make test

File: include/algorithms/math/abs_types.h [hunk 1]
```diff
@@ -97,17 +97,27 @@ namespace interface1
...
  * \brief %Input objects for the absolute value function
+ * \DAAL_DEPRECATED
  */
 class DAAL_EXPORT Input : public daal::algorithms::Input
 {
 public:
-    /** Default constructor */
+    /**
+     * Default constructor
+     * \DAAL_DEPRECATED_USE{ Batch() }
```
Comment (ShvetsKS) [inline include/algorithms/math/abs_types.h]: why do you recommend to use Batch instead of Input

File: include/algorithms/math/abs.h [hunk 2]
```diff
@@ -89,15 +96,19 @@ class DAAL_EXPORT Batch : public daal::algorithms::Analysis<batch>
...
+     */
+    DAAL_DEPRECATED Batch()
     {
         initialize();
     }
 
     /**
      * Constructs the absolute value function by copying input objects of another absolute value function
      * \param[in] other function to be used as the source to initialize the input objects of the absolute value function
+     * \DAAL_DEPRECATED_USE{ cloneImpl() }
```
Comment (ShvetsKS) [inline include/algorithms/math/abs.h]: we have to recommend interface2::Batch instead

File: include/algorithms/math/abs_types.h [hunk 3]
```diff
@@ -97,17 +97,27 @@ namespace interface1
...
+     */
     Input();
 
-    /** Copy constructor */
+    /**
+     * Copy constructor
+     * \DAAL_DEPRECATED_USE{ Batch(const Batch<algorithmFPType, method> &other) }
```
Comment (ShvetsKS) [inline include/algorithms/math/abs_types.h]: the same

File: include/algorithms/math/logistic.h [hunk 4]
```diff
@@ -89,15 +96,19 @@ class DAAL_EXPORT Batch : public daal::algorithms::Analysis<batch>
...
+     */
+    DAAL_DEPRECATED Batch()
     {
         initialize();
     }
 
     /**
      * Constructs the logistic function by copying input objects of another logistic function
      * \param[in] other function to be used as the source to initialize the input objects of the logistic function
+     * \DAAL_DEPRECATED_USE{ cloneImpl() }
```
Comment (ShvetsKS) [inline include/algorithms/math/logistic.h]: cloneImpl is deprecated also, just DAAL_DEPRECATED

File: include/algorithms/math/logistic_types.h [hunk 5]
```diff
@@ -86,17 +86,27 @@ namespace interface1
...
+    /**
+     * Default constructor
+     * \DAAL_DEPRECATED_USE{ Batch() }
+     */
     Input();
 
-    /** Copy constructor */
+    /**
+     * Copy constructor
+     * \DAAL_DEPRECATED_USE{ Batch() }
```
Comment (ShvetsKS) [inline include/algorithms/math/logistic_types.h]: fix this in all files

**Comment by owerbat** [comment]
make test

File: include/algorithms/math/abs_types.h (hunk 1 above)
Comment (owerbat) [reply include/algorithms/math/abs_types.h]: fixed

File: include/algorithms/math/abs.h (hunk 2 above)
Comment (owerbat) [reply include/algorithms/math/abs.h]: fixed

File: include/algorithms/math/abs_types.h (hunk 3 above)
Comment (owerbat) [reply include/algorithms/math/abs_types.h]: fixed

File: include/algorithms/math/logistic.h (hunk 4 above)
Comment (owerbat) [reply include/algorithms/math/logistic.h]: fixed

File: include/algorithms/math/logistic_types.h (hunk 5 above)
Comment (owerbat) [reply include/algorithms/math/logistic_types.h]: fixed

**Comment by ShvetsKS** [comment]
CI:
e9ec0389-94ad-f12b-8489-8cdcd4b723a1

**Pull request merged**